gensim/test/test_fasttext.py

```diff
@@ -1300,6 +1305,299 @@ def _run(self, fin):
+            temporary_file("roundtrip_file_to_file2.bin") as fpath2bin, \
+            temporary_file("roundtrip_file_to_file1.vec") as fpath1vec:  # noqa:F841
+
+            fpath1base = fpath1bin[:-4]
+            self._create_and_save_test_model(fpath1base, model_params, fasttext_cmd)
+            model = gensim.models.fasttext.load_facebook_model(fpath1bin)
+            gensim.models.fasttext.save_facebook_model(model, fpath2bin)
+
+            self._compare_fasttext_files(fpath1bin, fpath2bin)
+
+    def save_test_file_file_skipgram(self):
+        model_params = {"size": 10, "sg": 1, "seed": 42}
+        self._check_roundtrip_file_file(model_params)
+
+    def save_test_file_file_cbow(self):
+        model_params = {"size": 10, "sg": 0, "seed": 42}
+        self._check_roundtrip_file_file(model_params)
+
+
+class SaveFacebookFormatReadingTest(SaveFacebookFormatTest):
+    """
+    This class containts tests that check the following scenario:
+
+    + create fastText model
+    + save file tom model.bin
+    + retrieve word vectors from model.bin to stdout using fasttext Facebook utility
+    + compare vectors retrieved by Facebook utility with those obtained directly from gensim model
+
+    Requires env. variable FT_HOME to point to location of Facebook fastText binary
+    """
+
+    def _parse_wordvectors(self, text):
+        def _conv_line_to_array(line):
+            return np.array([float(s) for s in line.split()[1:]], dtype=np.float32)
+
+        return np.array([_conv_line_to_array(l) for l in text.splitlines()], dtype=np.float32)
+
+    def _get_wordvectors_from_fb_fastttext(self, fasttext_cmd, fasttext_fname, words):
+        cmd = fasttext_cmd + " print-word-vectors " + fasttext_fname
+        process = subprocess.Popen(cmd, stdin=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                shell=True)
+        words_str = '\n'.join(words)
+        out, err = process.communicate(input=words_str.encode("utf-8"))
+        return self._parse_wordvectors(out.decode("utf-8"))
+
+    def _check_load_fasttext_format(self, model_params):
+
+        ft_home = os.environ.get("FT_HOME", None)
+        if ft_home is None:
+            self.skipTest("FT_HOME env variable not set")
+
+        fasttext_cmd = os.path.join(ft_home, "fasttext")
+
+        with temporary_file("load_fasttext.bin") as fpath:
+            model = self._create_and_save_test_model(fpath, model_params)
+            wv = self._get_wordvectors_from_fb_fastttext(fasttext_cmd, fpath, model.wv.index2word)
+
+            for i, w in enumerate(model.wv.index2word):
+                diff = calc_max_diff(wv[i, :], model.wv[w])
+                # Because fasttext command line prints vectors with limited accuracy
+                self.assertLess(diff, 1.0e-4)
+
+    def test_load_fasttext_format_cbow(self):
+        model_params = {"size": 10, "min_count": 1, "hs": 1, "sg": 0,
```

Comment:
Hanging indent please, here and everywhere.

We don't use vertical indent in gensim at all (hard to maintain).

Reply:
Corrected, everywhere where I could spot it.

Reply:
PS. Have you considered using `black` formatter?

Perhaps it could alleviate such problems once and forever ;) (e.g. spacy I believe uses it).

Reply:
Yes, there was a discussion around using `black` a while back (1-2 years ago).  IIRC there was some issue, so we only implemented a PEP8 check.

But it's a good idea in general, we should revisit this.

Reply:
Personally, I'm against Black. My reasons:

1. I consider code formatting to be a form of expression (like art). I'd prefer to not have a computer program stomp all over it. It feels too authoritarian. This is the main reason.
2. I don't Black's style. Yes, our code base will be consistently formatted, but it will look like crap.